### ISSUE TITLE
Feature Proposal - More robust resuming?

### ISSUE THREAD

**Issue by dgcrouse** [jcjohnson/torch-rnn#131]
Right now when resuming from a checkpoint you have to remember many of the parameters that you set, which is difficult if you're working from old saved nets (esp. calculating the initial learning rate. Is there any interest in adding the capability to "resume from" a previous checkpoint using a JSON file saving all parameters and the _current_ learning rate?

**Comment by dgcrouse** [comment]
See my branch for an early prototype.

**Comment by dgcrouse** [comment]
Just noticed that JSON checkpoints save the initial parameters but don't use them? Perhaps we could just split this out into the resume JSON file. That way one contains progress and the other settings?

**Comment by dgcrouse** [comment]
See my fork for new proposed system. Two json checkpoints, one containing all options and current state needed to resume (except the model), another containing timing and loss function history of the network. The latter can be toggled on and off by an option.

**Comment by dgcrouse** [comment]
I'm going to close this and move it to the feature discussion we've just started.
